smslibrary/src/main/java/com/eis/smslibrary/SMSPeer.java

```diff
@@ -0,0 +1,152 @@
+package com.eis.smslibrary;
+
+import androidx.annotation.NonNull;
+
+import com.eis.communication.Peer;
+
+/**
+ * Implementation of {@link Peer} for SMS communication channel.
+ *
+ * @author Luca Crema, Marco Mariotto.
+ */
+public class SMSPeer implements Peer<String>, Comparable<SMSPeer> {
+
+    public static final int MAX_TELEPHONE_NUMBER_LENGTH = 20;
+    public static final int MIN_TELEPHONE_NUMBER_LENGTH = 3;
+
+    private String telephoneNumber;
+
+    /**
+     * @param telephoneNumber Address for the peer.
+     * @throws InvalidTelephoneNumberException If telephoneNumber check is not {@link TelephoneNumberState#TELEPHONE_NUMBER_VALID}.
+     */
+    public SMSPeer(String telephoneNumber) throws InvalidTelephoneNumberException{
+        TelephoneNumberState telephoneNumberState = SMSPeer.checkPhoneNumber(telephoneNumber);
+        if(telephoneNumberState != TelephoneNumberState.TELEPHONE_NUMBER_VALID)
+            throw new InvalidTelephoneNumberException(telephoneNumberState);
+
+        this.telephoneNumber = telephoneNumber;
+    }
+
+    /**
+     * Checks if the given phone number is valid.
+     *
+     * @param telephoneNumber the phone number to check.
+     * @return The {@link TelephoneNumberState} of the telephone number after the tests.
+     */
```

Comment:
If telephoneNumber is null throw NullPointerException

Reply:
And that is fine, it's implicit that parameter telephoneNumber must not be null

Reply:
Add it in the preconditions